Sources/SQLiteNIO/SQLiteData.swift

```diff
@@ -1,13 +1,32 @@
 import CSQLite
 import NIOCore
 
+#if _pointerBitWidth(_64)
+// We use Int instead of Int64 on 64 bit systems primarily for
+// backwards compatibility, since prior code versions have used Int here.
+public typealias SQLiteDataIntegerType = Int // 64-bit platform, Int = 64 bits
+#elseif _pointerBitWidth(_32)
+public typealias SQLiteDataIntegerType = Int64 // On 32-bit platforms, we want to use 64 bit integers.
+#else
+// If you hit errors here, you may simply need to add
+// a new architectural bit size above (eg. _128) when that
+// exists.
+//
+// Or if the following proposal for pointerBitWidth ever lands in a
+// published Swift version, then the above conditionals may
+// need adjusted:
+//
+// https://forums.swift.org/t/pitch-pointer-bit-width-compile-time-conditional/59572
+#error("Unsupported integer size")
+#endif
```

Comment:
sqlite-no has (obviously) not previously supported 32-bit runtime environments. I'm not familiar enough with Wasm to know whether 32-bit is an environment we should be expecting to encounter on a regular basis?